### DISCUSSION TITLE
How to ensure that symbols in a topic appear only for the first occurrence of the word?

### DISCUSSION THREAD

**Discussion by syeda-git** [facelessuser/pymdown-extensions#1659] [Q&A] [answered]
In my topic, the product name, say ABC, is used multiple times. I have the name ABC defined in a file so everyone in my team use the same name. We use the Snippet extension. The requirement is to have the registered trademark symbol appear only on the first occurrence of the product name.

So, if the source has:
**With ABC(r), you can do this and that. You can also do this with ABC(r).**

The output should be:
**With ABC®, you can do this and that. You can also do this with ABC.**

Is there a way to handle this case?

Syed

**Comment by facelessuser** [comment, marked as answer]
I'm not quite sure what this has to do with Snippets. I assume this question is actually more about SmartSymbols.

SmartSymbols is meant to translate certain syntax to symbols, in your case, any time it sees `(r)` to  `®`, assuming this symbol is [enabled in SmartSymbols](https://facelessuser.github.io/pymdown-extensions/extensions/smartsymbols/#options). If you do not want a symbol to be processed, you need to escape them, e.g., `\(r)`.

You are essentially asking for SmartSymbols to keep track of whether it has seen a given symbol, and then throw it away if it already has.  This kind of logic would not apply to most symbols, and unfortunately is so niche, and specific to your special needs, that it doesn't make sense as a generalized feature in Pymdown Extensions.

**Reply by facelessuser** [reply]
I will add, this doesn't mean it is impossible, surely you could run some post-processor script to remove extra "registered" symbols, but it is not something builtin to Pymdown Extensions, nor do I think it makes sense to.

**Reply by syeda-git** [reply]
Thanks. Can you please share what kind to script is acceptable? JavaScript, Perl, Python etc. Also, are there instructions how to include the script in the post-processor step?

**Reply by facelessuser** [reply]
I don't think there is a real answer to this. Post-processing can be done with anything as it is done after the Markdown process. Just pick what you are comfortable working with.

With that said, if your plan is to create a Python Markdown post-processing plugin, then I'd look at the Python Markdown extension API: https://python-markdown.github.io/extensions/api/#postprocessors.

Maybe your documentation system has a post-processing hook, then you could consider that as well.
